uber/reports.py

```diff
@@ -0,0 +1,48 @@
+from uber.common import *
+
+
+class ReportBase:
+    def write_row(self, row, out):
+        # plugins can override this hook
+        out.writerow(row)
+
+
+class PersonalizedBadgeReport(ReportBase):
+    """
+    Generate a CSV file which contains personalized badges with custom printed_names on them
+    """
+
+    def __init__(self, include_badge_nums=True):
+        self._include_badge_nums = include_badge_nums
+
+    def run(self, out, session, *filters, order_by=None, badge_type_override=None):
```

Comment:
(Finally going through the backlog of old reviews.)

The ``*whatever`` syntax in Python means "all other positional arguments".  By putting ``order_by=None`` and ``badge_type_override=None`` after ``*filters`` it makes them "keyword-only arguments".  For example, if I have a function with only regular arguments, I can call its parameters as either positional or keyword arguments:

```python
def f(x, y): pass

f(5, 6)      # positional parameters
f(5, y=6)    # mix of positional and keyword
f(x=5, y=6)  # both keyword arguments
f(y=6, x=5)  # order doesn't matter for keyword arguments
```

But if I used the ``*`` syntax, then anything after those arguments would be keyword-only, e.g. if I said

```python
def f(*, x=None, y=None): pass

f(5, 6)      # invalid, will raise an exception
f(x=5, y=6)  # this is fine
```